raster/r.kappa/calc_metrics.c

```diff
@@ -0,0 +1,214 @@
+        metrics->rowsum[cndx] = t_row;
+    }
+    if (metrics->obs == 0) {
+        metrics->total_acc = 0.0;
+        metrics->kappa = na_value;
+        metrics->kappa_var = na_value;
+        return;
+    }
+
+    /* Calculate kappa values */
+    pi = (double *)G_calloc(ncat, sizeof(double));
+    pj = (double *)G_calloc(ncat, sizeof(double));
+    pii = (double *)G_calloc(ncat, sizeof(double));
+    metrics->cond_kappa = (double *)G_calloc(ncat, sizeof(double));
+    metrics->user_acc = (double *)G_calloc(ncat, sizeof(double));
+    metrics->prod_acc = (double *)G_calloc(ncat, sizeof(double));
+
+    for (i = 0; i < ncat; i++) {
+        for (l = 0; l < nstats; l++) {
+            if (Gstats[l].cats[0] == rlst[i]) {
+                pi[i] += Gstats[l].count;
+            }
+
+            if (Gstats[l].cats[1] == rlst[i]) {
+                pj[i] += Gstats[l].count;
+            }
+
+            if ((Gstats[l].cats[0] == Gstats[l].cats[1]) &&
+                (Gstats[l].cats[0] == rlst[i])) {
+                pii[i] += Gstats[l].count;
+            }
+        }
+        metrics->correct += pii[i];
+    }
+
+    metrics->total_acc = 100. * metrics->correct / metrics->obs;
+
+    /* turn observations into probabilities */
+    for (i = 0; i < ncat; i++) {
+        pi[i] = pi[i] / metrics->obs;
+        pj[i] = pj[i] / metrics->obs;
+        pii[i] = pii[i] / metrics->obs;
+        if (pi[i] == 0)
+            metrics->user_acc[i] = na_value;
+        else
+            metrics->user_acc[i] = 100 * (pii[i] / pi[i]);
+        if (pj[i] == 0)
+            metrics->prod_acc[i] = na_value;
+        else
+            metrics->prod_acc[i] = 100 * (pii[i] / pj[i]);
+        /* theta 1 */
+        p0 += pii[i];
+        /* theta 2 */
+        pC += pi[i] * pj[i];
+    }
+    if (pC != 1)
+        metrics->kappa = (p0 - pC) / (1 - pC);
+    else
+        metrics->kappa = na_value;
+
+    /* conditional kappa */
+    for (i = 0; i < ncat; i++) {
+        if (pi[i] == 0 || (pi[i] == 1 && pj[i] == 1))
+            metrics->cond_kappa[i] = na_value;
+        else
+            metrics->cond_kappa[i] =
+                (pii[i] - pi[i] * pj[i]) / (pi[i] - pi[i] * pj[i]);
+        inter1 += pii[i] * pow(((1 - pC) - (1 - p0) * (pi[i] + pj[i])), 2.);
+    }
+
+    /* kappa variance */
+    for (l = 0; l < nstats; l++) {
+        if (Gstats[l].cats[0] != Gstats[l].cats[1]) {
+            for (i = 0; i < ncat; i++) {
+                if (Gstats[l].cats[0] == rlst[i])
+                    a_i = i;
+                if (Gstats[l].cats[1] == rlst[i])
+                    b_i = i;
+            }
+            inter2 +=
+                Gstats[l].count * pow((pi[a_i] + pj[b_i]), 2.) / metrics->obs;
+        }
+    }
+    metrics->kappa_var = (inter1 + pow((1 - p0), 2.) * inter2 -
+                          pow((p0 * pC - 2 * pC + p0), 2.)) / pow((1 - pC),
+                                                                  4.) /
+        metrics->obs;
```

Comment:
GNU indent makes a lousy job here. This is what clang-format would produce:
```c
    metrics->kappa_var = (inter1 + pow((1 - p0), 2.) * inter2 -
                          pow((p0 * pC - 2 * pC + p0), 2.)) /
                         pow((1 - pC), 4.) / metrics->obs;
```

(just a little push for #2272).

Reply:
Yeah, clang-format version looks better. Pity that #2272 contains both – .clang-format and code formatting as merging just configuration file would be easier.

Reply:
As I mentioned in the PR description:

> Note: only the addition of the .clang-format is the ultimate goal for this PR. The added commit with all the formatted source files are for illustration only.

The issue has been stalled somehow, see also my question https://github.com/OSGeo/grass/pull/2272#issuecomment-1075676212 . Should probably bring the question to dev-ML.

Reply:
Sorry for missing the note.
Yes, you should write to dev ML with clean statement – if no objections are made till XX date, we'll go with Stroustrup.